exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/conv/DateEpochBEConvertFrom.java

```diff
@@ -41,6 +41,7 @@ public void eval() {
 
     in.buffer.readerIndex(in.start);
     long epochMillis = Long.reverseBytes(in.buffer.readLong());
+    in.buffer.readerIndex(0);
```

Comment:
Imagine this case: 
A downstream operator asks Project for producing two columns, convert_to(col, ...) and col. 

Due to the current implementation of convert_to(), after the the convert_to(col, ...) evaluation is done, the readIndex in col's Drillbuf will be pointed at the end. 

Then after the receiver uses RecordBatchLoader to read this Drillbuf, IOOB will be thrown